crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/service/FullMeansAssessmentService.java

```diff
@@ -3,22 +3,66 @@
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Service;
+import uk.gov.justice.laa.crime.meansassessment.dto.MeansAssessmentDTO;
 import uk.gov.justice.laa.crime.meansassessment.model.common.ApiCreateMeansAssessmentRequest;
-import uk.gov.justice.laa.crime.meansassessment.model.common.ApiCreateMeansAssessmentResponse;
-import uk.gov.justice.laa.crime.meansassessment.validation.service.MeansAssessmentValidationService;
+import uk.gov.justice.laa.crime.meansassessment.staticdata.entity.AssessmentCriteriaEntity;
+import uk.gov.justice.laa.crime.meansassessment.staticdata.enums.CurrentStatus;
+import uk.gov.justice.laa.crime.meansassessment.staticdata.enums.FullAssessmentResult;
+
+import java.math.BigDecimal;
 
 @Service
 @RequiredArgsConstructor
 @Slf4j
-public class FullMeansAssessmentService {
+public class FullMeansAssessmentService implements AssessmentService {
+
+    private final AssessmentCriteriaChildWeightingService childWeightingService;
+
+    public MeansAssessmentDTO execute(BigDecimal expenditureTotal, ApiCreateMeansAssessmentRequest meansAssessment, AssessmentCriteriaEntity assessmentCriteria) {
+        log.info("Create full means assessment - Start");
+        BigDecimal adjustedLivingAllowance = getAdjustedLivingAllowance(meansAssessment, assessmentCriteria);
+        BigDecimal totalDisposableIncome = getDisposableIncome(meansAssessment, expenditureTotal, adjustedLivingAllowance);
+        CurrentStatus status = meansAssessment.getAssessmentStatus();
+        log.info("Full means assessment calculation complete for Rep ID: {}", meansAssessment.getRepId());
+        return MeansAssessmentDTO
+                .builder()
+                .currentStatus(status)
+                .fullAssessmentResult(
+                        status.equals(CurrentStatus.COMPLETE) ? getResult(
+                                totalDisposableIncome, assessmentCriteria
+                        ) : FullAssessmentResult.NONE
+                )
+                .adjustedLivingAllowance(adjustedLivingAllowance)
+                .totalAggregatedExpense(expenditureTotal)
+                .totalAnnualDisposableIncome(totalDisposableIncome).build();
+    }
 
-    public ApiCreateMeansAssessmentResponse createFullAssessment(ApiCreateMeansAssessmentRequest apiCreateMeansAssessmentRequest) {
+    BigDecimal getDisposableIncome(ApiCreateMeansAssessmentRequest meansAssessment, BigDecimal expenditureTotal, BigDecimal adjustedLivingAllowance) {
+        return meansAssessment.getInitTotalAggregatedIncome().subtract(
```

Comment:
Good catch! 

I think rather than updating this part of the code it would be wiser to split the existing request schema into two. The same probably applies to quite a few of the fields and right now we're missing out on one of the main benefits of using a code generation tool in the first place, which is to provide automatic validation. 

I'll raise a separate Jira ticket to address this.